File: Project Main Ui/src/app/Models/OrderDetails.ts
export interface Order {
    id: string;
    userId: number;
    name: string;
    email: string;
    address:string;
    total:number;
    products: Product[];
  }
  
  export interface Product {
    name: string;
    price: number;
    quantity: number;
  }
  

File: MongoMicroservice/Models/OrderDetails.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;

namespace MongoMicroservice.Models
{
    public class OrderDetails
    {

        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string id { get; set; }
        public int userId { get; set; }
        
        public string name { get; set; }

       
        public string email { get; set; }

        public string address { get; set; }

        public int total { get; set; }
        public List<Product> products { get; set; }
    }

    public class Product
    {
        public string name { get; set; }
        public decimal price { get; set; }
        public int quantity { get; set; }
    }
}
